runtime/onert/core/include/backend/basic/BackendContextHelpers.h

```diff
@@ -182,9 +182,24 @@ template <typename T_BackendContext> ITensorRegistry *genTensors(T_BackendContex
   const ir::Graph &graph = *ctx.graph();
   auto tensor_builder = ctx.tensor_builder;
 
+  // process source tensors for shared memory at first
+  std::vector<ir::OperandIndex> registered_source_ind;
+  for (const auto &[_, source_ind] : tensor_builder->getSharedMemoryOperandIndexes())
+  {
+    if (ctx.external_operands().contains(source_ind))
+      continue;
+    if (tensor_builder->isRegistered(source_ind)) // some tensors can have the same source
+      continue;
+    tensor_builder->registerTensorInfo(source_ind, graph.operands().at(source_ind).info());
+    registered_source_ind.emplace_back(source_ind);
+  }
+
   graph.operands().iterate([&](const ir::OperandIndex &ind, const ir::Operand &obj) {
     if (ctx.external_operands().contains(ind))
       return;
+    if (std::find(std::begin(registered_source_ind), std::end(registered_source_ind), ind) !=
+        std::end(registered_source_ind)) // skip tensors already registered
+      return;
```

Comment:
While reviewing https://github.com/Samsung/ONE/pull/14228, I re-read this PR.

I'm a bit confused about this part. 
In `genTensors()`, Is it sufficient just to register the `source_ind` first? 

-----------------------------------------

I thought that from the pair of `(shared_ind, source_ind)`, this draft try to allocate only `source operand` and avoid allocating `shared operand`. I failed to understand how this code removes the allocation of `shared operand`. 

I guess there is sth I missed. Could you help me to understand?

Reply:
Note that `registerTensorInfo` is responsible just for `buildTensor` (both `non-const` and `const`).

The second step is calling `allocateNonconsts` where we are calling `tensor->setBuffer`. Just at this point we are passing the same buffer for source tensor and shared tensor. It requires of course special handling of such memory lifetime. The lifetime is controlled by `StaticTensorManager::claimPlan` called during first use of memory buffer and `StaticTensorManager::releasePlan` called during the last use of memory buffer (the graph has to be topologically sorted).

Conclusion - we are creating tensors both for `source_operand` and `shared_operand` - they are just share memory buffer passed by `setBuffer` method.

Registering source tensors at the beginning here is needed to proper handling cases where a source tensor is constant - in such a case the shared tensor tensors has to be also a constant (has `ExternalTensor` type).
Without this additional code here we have no guarantee that source operands will be processed at first.

Reply:
Aha, Thank you a lot for your kind explanation :+1: